### DISCUSSION TITLE
Issue on iOS

### DISCUSSION THREAD

**Discussion by RuellePaul** [gilbarbara/react-spotify-web-playback#152] [General]
Hi,

The lib is working perfectly for my app, on all browsers, except iOS (Safari). Safari on desktop Mac is working.

Do you have any clues why, and how it could be fixed ?

**Comment by gilbarbara** [comment]
Hey @RuellePaul 

That's too vague.

What is happening?
What version of Safari and the library are you using?
Does it work in other iOS browsers, like Chrome?

**Comment by RuellePaul** [comment]
The problem happens for all users who tried the app on iOS/Safari (iPhone 12/13/14).

Not having an iphone myself, I'll find out about their version of Safari and if the site works on another version.

The app (melodymaster.io) is online, in developer mode for Spotify. I added you today on developers.spotify.com so you can try it yourself, if you have an iPhone.

Users click on the "play" button, but nothing happens for them. Their spotify token is valid though. I haven't logged the error from the CallbackState yet, I'll take care of it and give you the information within a week :).

This is my code for the SpotifyPlayer :

```
<SpotifyPlayer
    autoPlay
    callback={async ({isPlaying, errorType}) => {
        setIsPlaying(isPlaying);

        if (errorType === 'authentication_error') modal.open('spotify_access_token_expired');
    }}
    name="Melody Master"
    token={spotifyAccessToken}
    uris={[`spotify:track:${saloon.games[saloon.games.length - 1].song.id}`]}
    showSaveIcon
    styles={{
        activeColor: palette.primary.main,
        altColor: palette.text.primary,
        bgColor: palette.background.main,
        color: palette.text.primary,
        errorColor: palette.error.main,
        loaderColor: palette.primary.main,
        sliderColor: palette.primary.main,
        sliderHandleColor: palette.primary.main
    }}
/>
```

Finally, I will also find out if this works for other browsers on iOS.

Thanks a lot for your help,

Paul

**Comment by RuellePaul** [comment]
It happens only for Safari on iOS.

**Comment by gilbarbara** [comment]
Hey @RuellePaul 

Some changes in the latest Safari versions regarding autoplay policies impacted Spotify's Web Playback SDK.

There's an ongoing [discussion](https://community.spotify.com/t5/Spotify-for-Developers/Web-Playback-SDK-Playing-song-directly-in-browser-issues-IOS/m-p/5538195/) in their community forum about it, and I spent 12 hours trying to get around the problem besides some initial success with a simple setup, I couldn't get this library to work yet.
I'll try to "fix" it again when I have time, but I think the SDK needs to fix this.

**Reply by RuellePaul** [reply]
Thanks for your answer, I understand better now. It's a problem that directly concerns the Spotify SDK...

Good luck in solving the problem, I can't wait to see it work on iOS :)

**Reply by gilbarbara** [reply]
Spotify finally fixed the issue in the SDK.
The player is working in Safari again. Can you confirm?
